### ISSUE TITLE
Questions about the GOOD-motif dataset

### ISSUE THREAD

**Issue by Artimislyy** [divelab/GOOD#19]
1.What is the difference between the id_val_dataset and val_dataset?
2.what does train_dataset.data.env_id mean?

**Comment by CM-BF** [comment]
Hi Yangyang,

Thank you for your questions.

1. The `id_val_dataset` is the in-domain split in GOOD-Motif which shares similar distribution with the training set. In contrast, the `val_dataset` is the out-of-domain split in GOOD-Motif that consists of different distributions from the training set, *i.e.*, different base graphs.
2. The `train_dataset.data.env_id` indicates the environment labels of samples in the dataset, which serves the same purpose as the environment partitions in Invariant Risk Minimization (IRM). For more information, please refer to our paper.

Please let me know if you have any further questions! :smile:

Best regards,
Shurui Gui
